plugins/modules/rma_workflow_manager.py

```diff
@@ -559,24 +559,84 @@ def rma_device_replacement_pre_check(self):
-            return self
+        if not self.have["is_pnp_replacement_device"]:
+            if self.have["replacement_device_reachability_status"] != "Reachable":
+                self.msg = "The replacement device is not reachable. Unable to proceed with the RMA device replacement."
+                self.log(self.msg, "ERROR")
+                self.status = "failed"
+                return self
+            self.log("The replacement device '{0}' is reachable.".format(self.have.get("replacement_device_name")), "DEBUG")
 
-        self.log("The replacement device '{0}' is reachable.".format(self.have.get("replacement_device_name")), "DEBUG")
         return self
 
+    def pnp_device_exists(self, identifier, identifier_type):
+        """
+        Check if a pnp device exists in Cisco Catalyst Center and return its device ID, device_name, serial_number and platform_id.
+        Parameters:
+            - self (object): An instance of the class containing the method.
+            - identifier (str): The identifier of the device to check.
+            - identifier_type (str): The type of identifier (name, ip_address, or serial_number).
+        Returns:
+            - dict: A dict containing the device ID, device_name, serial_number and platform_id if the device
+              is found or empty dict if device not found.
+        Description:
+            This method queries Cisco Catalyst Center to check if a specified device exists based on the provided identifier.
+            It constructs the appropriate query parameters based on the identifier type (hostname, IP address, or serial number).
+            The method then sends a request to Cisco Catalyst Center using the 'get_device_list' function.
+            If the device is found and both ID and serial number are available, it returns these as a tuple.
+            If the device is not found, lacks necessary information, or if an error occurs during the process,
+            it logs an appropriate error message and returns empty dict.
+            This method is used to verify the existence of both faulty and replacement devices in the RMA workflow.
+        """
+        params = {}
+        if identifier_type.endswith("_name"):
+            params["hostname"] = identifier
+        elif identifier_type.endswith("_serial_number"):
+            params["serialNumber"] = identifier
+        else:
+            self.log("Invalid identifier type provided", "ERROR")
+            return {}
+
+        try:
+            response = self.dnac._exec(
+                family="device_onboarding_pnp",
+                function='get_device_list',
+                op_modifies=False,
+                params=params
+            )
+            self.log("Received API response from 'get_device_list': {0}".format(self.pprint(response)), "DEBUG")
+            device_param_list = {}
+
+            if response and response[0]:
+                if len(response[0]) > 0:
+                    device = response[0]
```

Comment:
device_info = device.get('deviceInfo', {})

Reply:
Updated the code accordingly.